Extend save when

WenzDaniel opened the pull request:
## What does the code in this PR do / what does it improve?
Uses https://github.com/AxFoundation/strax/pull/631 to store pulse count information without storing records. Changes lineage of low level plugin so only merge when larger reprocessing campaign is planned. Also requires to bump strax requirement. 

- [x] Bump strax requirement once https://github.com/AxFoundation/strax/pull/631 is merged.
- [x] Bump strax requirement once https://github.com/AxFoundation/strax/pull/648 is released

JoranAngevaare commented on straxen/plugins/nveto_recorder.py:
```diff
@@ -51,11 +51,14 @@ class nVETORecorder(strax.Plugin):
     rechunk_on_save = True
-    save_when = strax.SaveWhen.TARGET
+    save_when = {'raw_records_coin_nv': strax.SaveWhen.TARGET,
+                 'lone_raw_records_nv': strax.SaveWhen.EXPLICIT,
```
I would consider also setting this to TARGET? It's not that easy to compute

JoranAngevaare commented on straxen/plugins/pulse_processing.py:
```diff
@@ -134,7 +134,10 @@ class PulseProcessing(strax.Plugin):
     data_kind = {k: k for k in provides}
-    save_when = strax.SaveWhen.TARGET
+    save_when = {'records': strax.SaveWhen.TARGET,
+                 'veto_regions': strax.SaveWhen.EXPLICIT,
```
```suggestion
                 'veto_regions': strax.SaveWhen.TARGET,
```

JoranAngevaare reviewed (changes requested):
@WenzDaniel , there is no reason to bump the version of these low level data types right?

Additionally, should the `save_when` be immutable just to be sure?

WenzDaniel commented:
Okay I addressed all your comments and changed the code accordingly. I did not have any strong opinions. Regarding the version you are absolutely right :D  Usually I forget it when I have to and now I was too eager :D

JoranAngevaare reviewed (approved):
Very nice! Thanks

JoranAngevaare commented:
@WenzDaniel , do you want to have a look at the tests?

JoranAngevaare commented:
some tweaks are needed, we should not forget to bump the version https://github.com/AxFoundation/strax/pull/648

JoranAngevaare commented:
@WenzDaniel https://github.com/XENONnT/straxen/pull/879/commits/8c62125688590594f4f0938f922e3b4e7eee3c2e finished successfully :)
